internal/service/probe.go

```diff
@@ -130,17 +128,58 @@ func (ps *ProbeService) Delete(name string) error {
 		return ErrProbeNotFound
 	}
 
-	mapProbe.finish <- true
-	delete(ps.probes, mapProbe.Name)
+	return ps.deleteProbe(*mapProbe)
+}
 
+func (ps *ProbeService) insertProbe(probe model.Probe) error {
+	dbClient, err := database.GetClient()
+	if err != nil {
+		return err
+	}
+	_, err = dbClient.InsertProbe(probe, func(probe model.Probe) error {
+		probe.Finish = make(chan bool, 1)
+		ps.probes[probe.Name] = &probe
+		return nil
+	})
+	return err
+}
+
+func (ps *ProbeService) deleteProbe(probe model.Probe) error {
+	dbClient, err := database.GetClient()
+	if err != nil {
+		return err
+	}
+	err = dbClient.DeleteProbeByID(probe.ID, func() error {
+		probe.Finish <- true
+		delete(ps.probes, probe.Name)
+		return nil
+	})
+	return err
+}
+
+func (ps *ProbeService) initWithStoredProbes() error {
+	dbClient, err := database.GetClient()
+	if err != nil {
+		return err
+	}
+	probes, err := dbClient.ReadAllProbes()
+	if err != nil {
+		return err
+	}
+	for _, probe := range probes {
+		copy := probe.Copy()
+		copy.Finish = make(chan bool, 1)
+		ps.probes[copy.Name] = &copy
+		go ps.run(&copy)
+	}
 	return nil
 }
```

Comment:
I created these functions to call persistence layer because it provides the callback function to use in the transaction, which could be seen as service related.
I don't see any issues for moving this fully to the persistence layer, as it will always do the same actions on insert/delete/readAll anyways.

Reply:
Actually I'm not sure why I went this way, I will remove these callback and keep the service part inside the service